src/flatten.js

```diff
@@ -0,0 +1 @@
+module.exports = Array.prototype.concat.bind([])
```

Comment:
This let's us use `[].concat` to ensure arguments passed into `transaction`, `push` and `prepare` are arrays. There's no reason to use a new array every single time.